src/store.rs

```diff
@@ -19,20 +21,28 @@ pub enum Visibility {
 }
 
 /// Implementable by ipld storage providers.
-pub trait ReadonlyStore: Clone {
+pub trait ReadonlyStore<C = Codec, H = MultihashCode>: Clone
+where
+    C: Into<u64> + TryFrom<u64> + Copy,
+    H: Into<u64> + TryFrom<u64> + Copy,
+{
     /// Returns a block from the store. If the block is not in the
     /// store it fetches it from the network and pins the block. This
     /// future should be wrapped in a timeout. Dropping the future
```

Comment:
>  This future should be wrapped in a timeout.

Is no longer true. The future returned by the store should timeout if it makes sense.

Reply:
That's unrelated to this change => separate PR.